src/engine/srv.c

```diff
@@ -602,27 +608,38 @@ dss_start_one_xstream(hwloc_cpuset_t cpus, int xs_id)
 	 * as it is only for EC/checksum/compress offloading.
 	 */
 	if (dss_helper_pool) {
-		comm = (xs_id == 0) || (xs_id >= dss_sys_xs_nr &&
-				xs_id < (dss_sys_xs_nr + 2 * dss_tgt_nr));
+		comm =  (xs_id == 0) || /* DSS_XS_SYS */
+			(xs_id == 1) || /* DSS_XS_SWIM */
```

Comment:
Per the commit msg "Let's assign SWIM to an independent xstream to avoid timeout, if there are extra cores avabile, otherwise the swim ULT will be still assigned in the same core(xstream) as the system xstream(0)".

Seems that whether or not the xs_id 1 is for SWIM (or original's drpc XS), and whether or not create cart context for the xs_id 1 should be configurable? but the logic here always assume the xs_id 1 must for SWIM and always create cart_context?

Reply:
reworked the logic.